packages/jellyfish-api-core/src/category/loan.ts

```diff
@@ -599,10 +602,16 @@ export interface TakeLoanMetadata {
   to?: string
 }
 
+export interface TokenPaybackAmount {
+  dToken: string
+  amounts: string | string[] // amount@symbol
+}
+
 export interface PaybackLoanMetadata {
   vaultId: string
-  amounts: string | string[] // amount@symbol
+  amounts?: string | string[] // amount@symbol
   from: string
+  loans?: TokenPaybackAmount[]
```

Comment:
Since there must be either a `amounts` or `loans` field, and not the absence of both, I would suggest something like so to add constraints.

```typescript
export interface TokenPaybackAmount {
  dToken: string
  amounts: string | string[] // amount@symbol
}


export interface PaybackLoanMetadataWithAmounts {
  vaultId: string
  amounts: string | string[] // amount@symbol
  from: string
}

export interface PaybackLoanMetadataWithTokenPaybackAmount {
  vaultId: string
  from: string
  loans: TokenPaybackAmount[]
}
export type PaybackLoanMetadata = PaybackLoanMetadataWithAmounts | PaybackLoanMetadataWithTokenPaybackAmount
```

Reply:
Good point, thanks for the suggestion